lib/index.js

```diff
@@ -28,26 +43,53 @@ NamedURLResolverClass.prototype.escape = function(string) {
 
 NamedURLResolverClass.prototype.resolve = function(name, params) {
     if(name && (name in this.routesMap)) {
-        if(!params) params = {};
-
         var routePath = this.routesMap[name];
-        for(var paramName in params) {
-            if(params.hasOwnProperty(paramName)) {
-                let paramRegex = new RegExp('(/|^):' + paramName + '(/|$)');
-                let paramValue = this.escape('' + params[paramName]);
-                routePath = routePath.replace(paramRegex, '$1' + paramValue + '$2');
+
+        if(!params) {
+            return routePath;
+        }
+
+        for (var paramName in params) {
+            if (params.hasOwnProperty(paramName)) {
+                var paramValue = params[paramName];
+        
+                if (paramName === "splat") { // special param name in RR, used for "*" and "**" placeholders
+                    paramValue = toArray(paramValue); // when there are multiple globs, RR defines "splat" param as array.
+                    var i = 0;
+                    routePath = routePath.replace(reSplatParams, (match) => {
+                        var val = paramValue[i++];
+                        // don't escape slashes for double star, as "**" considered greedy by RR spec
+                        return val == null ? "" : this.escape('' + val, match === "*");
+                    });
+                } else {
+                    // Rougly resolve all named placeholders.
+                    // Cases: 
+                    // - "/path/:param"
+                    // - "/path/(:param)"
+                    // - "/path(/:param)"
+                    // - "/path(/:param/):another_param"
+                    // - "/path/:param(/:another_param)"
+                    // - "/path(/:param/:another_param)"
+                    var paramRegex = new RegExp('(\/|\\(|\\)|^):' + paramName + '(\/|\\)|\\(|$)');
+                    var replacement = '$1' + this.escape('' + paramValue) + '$2';
+                    routePath = routePath.replace(paramRegex, replacement);
+                }
             }
         }
-        return routePath;
+        return routePath
+            // Remove braces around resolved optional params (i.e. "/path/(value)")
```

Comment:
As we have a "recursive" tokens in string, a typical solution for such case is to introduce an additional tokens level.
For example (without any changes in `paramRegex`):
```js
var tokens = {};
...
routePath = routePath.replace(paramRegex, (match, g1, g2) => {
     tokens[paramName] = paramValue;
     return `${g1}@${paramName}@${g2}`; // `@` as example of additional token
 });
...
return routePath
            .replace(reResolvedOptionalParams, "$1")
            .replace(reUnresolvedOptionalParams, "")
            // after everything related to RR syntax is removed, insert actual values:
            .replace(/@(.*?)@/, (match, name) => tokens[name])
            .replace(reRepeatingSlashes, "/");
```
This works with things like `id(5)`.

Also, shouldn't we use `encodeURIComponent` for escaping values?